app/serializers/song_serializer.rb

```diff
@@ -1,5 +1,11 @@
 class SongSerializer < ActiveModel::Serializer
-  attributes :id, :title, :album, :duration
+  attributes :id, :title, :album, :duration, :full_title
+
   embed :ids
+
   has_one :artist
+
+  def duration
+    object.duration_in_minutes_and_seconds
```

Comment:
Since your `Song` model defines the `duration` attribute, it could be confusing to have the JSON return `duration` that is not that model attribute.

What we often do is either

* duration_seconds shows Song.duration, ie the integer value
* duration show the formatted value

or

* duration is the integer
* formatted_duration or ...
* duration_in_minutes shows the formatted string

Could see reasons why in the JSON that the mobile app using this data would find both values useful: one to present (5:13) and one to do calcs (played 13 of 313 seconds = 300 seconds remaining)